src/Filter/CheckboxFilter/CheckboxFilter.tsx

```diff
@@ -73,31 +86,48 @@ function CheckboxFilterPopper({
   name,
   onChange,
   options,
-  value
-}) {
+  value,
+  disabled
+}: Omit<CheckboxFilterProps, "variant">) {
   return (
     <Autocomplete
       limitTags={limitTags}
       multiple
-      onChange={(e, newValue) => onChange(newValue)}
+      onChange={(_e, newValue) => onChange(newValue)}
       options={options}
       renderInput={params => (
         <TextField {...params} label={label} name={name} />
       )}
       renderOption={Option}
       value={value}
+      disabled={disabled}
     />
   );
 }
 
 // renderer for a checkbox option
-function Option(props, option, { selected }) {
+function Option(
+  props: HTMLAttributes<HTMLLIElement>,
+  option: string,
+  { selected }: { selected: boolean },
+  { disabled }: Pick<CheckboxFilterProps, "disabled">
```

Comment:
I am not sure I understand why `selected` and `disabled` are wrapped in an object. I think either all are wrapped together, or none are wrapped.

Reply:
They are not wrapped, they are destructured. The 3rd and 4th input args are objects. What we're doing here is just pulling out the selected and disabled fields from the objects in those 3rd and 4th arg positions.

Reply:
You can refer to the MUI document for further explanation on the 4 args. https://mui.com/material-ui/api/autocomplete/

<img width="955" alt="image" src="https://github.com/user-attachments/assets/4ba6fe88-2ff4-4b2d-996c-8bf5daf3368d">